ui_framework/components/form/color_picker.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { ChromePicker } from 'react-color';
+
+export class KuiColorPicker extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showColorSelector: false,
+    };
+  }
+
+  closeColorSelector = () => {
+    this.setState({ showColorSelector: false });
+  };
+
+  toggleColorSelector = () => {
+    this.setState({ showColorSelector: !this.state.showColorSelector });
+  };
+
+  handleColorSelection = (color) => {
+    this.props.onChange(color.hex);
+  };
+
+  render() {
```

Comment:
Here's what the render method looks like using the above classes. I moved the "toggle" click handler onto the root element so the user can click the entire label and swatch instead of just the swatch, since that's a pretty small click area.

```jsx
  render() {
    const color = this.props.color || '#ffffff';
    const classes = classNames('kuiColorPicker', this.props.className);
    return (
      <div
        className={classes}
        aria-label={ this.props['aria-label'] }
        data-test-subj={ this.props['data-test-subj'] }
        onClick={ this.toggleColorSelector }
      >
        <div className="kuiColorPicker__preview">
          <div
            className="kuiColorPicker__swatch"
            aria-label="Select a color"
            data-test-subj="colorSwatch"
            style={{ background: this.props.color }}
          />
          <div
            className="kuiColorPicker__label"
            aria-label={`Color selection is ${color}`}
          >
            { color }
          </div>
        </div>
        {
          this.state.showColorSelector ?
            <div className="kuiColorPickerPopUp" data-test-subj="colorPickerPopup">
              <div className="kuiColorPickerPopUpOverlay" onClick={ this.closeColorSelector } />
              <ChromePicker
                color={ color }
                disableAlpha={ true }
                onChange={ this.handleColorSelection }
              />
            </div>
            : null
        }
      </div>
    );
  }
```